zeebe/feel/src/main/java/io/camunda/zeebe/feel/impl/FeelFunctionProvider.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.feel.impl;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.camunda.feel.context.JavaFunction;
+import org.camunda.feel.context.JavaFunctionProvider;
+import org.camunda.feel.syntaxtree.Val;
+import org.camunda.feel.syntaxtree.ValDayTimeDuration;
+import org.camunda.feel.syntaxtree.ValError;
+import org.camunda.feel.syntaxtree.ValNull$;
+import org.camunda.feel.syntaxtree.ValNumber;
+import org.camunda.feel.syntaxtree.ValString;
+import org.camunda.feel.syntaxtree.ValYearMonthDuration;
+
+public class FeelFunctionProvider extends JavaFunctionProvider {
+  private static final Map<String, List<JavaFunction>> FUNCTIONS =
+      Map.of("cycle", List.of(CycleFunction.INSTANCE, CycleInfiniteFunction.INSTANCE));
+
+  @Override
+  public Optional<JavaFunction> resolveFunction(final String functionName) {
+    throw new UnsupportedOperationException();
```

Comment:
❓ Last question from my end: do we anticipate this to become an issue? 😄 Do we know when this is called? Will this ever be called for our function provider? 😅

Reply:
It's definitely a bit ugly but the way we override methods here means that this should never be called (until someone changes `JavaFunctionProvider` of course :sweat_smile:)